plugins/radare2/radare2_main.ml

```diff
@@ -31,8 +31,13 @@ let provide_radare2 file =
     let extract name json = Yojson.Basic.Util.member name json in
     try
      let symbol_list = Yojson.Basic.Util.to_list (R2.with_command_j "isj" file) in
+     let strip str = let open String in
+      match chop_prefix str ~prefix:"sym.imp." with
+      | Some str -> str
+      | None -> str 
+     in
      List.fold symbol_list ~init:() ~f:(fun () symbol -> 
-     accept (Yojson.Basic.Util.to_string (extract "name" symbol)) (Yojson.Basic.Util.to_int (extract "vaddr" symbol) |> Z.of_int)
+     accept (Yojson.Basic.Util.to_string (extract "name" symbol) |> strip) (Yojson.Basic.Util.to_int (extract "vaddr" symbol) |> Z.of_int)
```

Comment:
Also, missed on the first review, but `Yojson.Basic.Util.to_int` will fail on addresses that do not fit into OCaml integers. We shall not assume that addresses will fit into the host OCaml's `int` data type (which could be as small as 31 bit only), neither we shall assume that they will fit into int32 or even int64 (though the latter we do assume in the LLVM backend, but mostly because it is assumed by LLVM itself). I would suggest extracting the string and then use an appropriate `Z.of_string*` function to parse it.

Reply:
Sadly this would not be possible, because the `vaddr` here is of variant `Int (i : int) (representing a Json value holding an ocaml int) which could only be extracted by Yojson.Basic.Util.to_int, this is rather a radare2-ocaml (or even for yojson) implementation detail.

Reply:
This is a critical flaw on the `radare2-ocaml` side, which should be using `Yojson.Safe` instead of basic. On our side, we have the following options, 
1. submit a PR to radare2-ocaml and switch it to the Safe module. This will break the interface and we will have to bump their version. But without that this bindings are barely usable for the real-world binaries. So we anyway shall do this, irrelevant to this PR.

2. Ignore the json output and switch to the text parsing. The pros is that it will be faster, the cons is that I am not sure how stable is the output format of radare2. 

@XVilka can you advise us on this, is option 2 viable at all? Or maybe there are other options, like other output formats that are more reliable?

Reply:
Besides, we also have an option to fix radare2-ocaml without breaking the interfaces (or at least backport the fix to the 1.x versions), we can use [Safe.to_basic], which is safe and translates `Intlit x` to `String x`. The cost is performance  (yet another format translation), but the benefits are obvious, preserved interfaces. The downstream code should still be updated to expect `String`s in places where `Int`s do not fit the OCaml integers.

[1]: https://ocaml-community.github.io/yojson/yojson/Yojson/Safe/index.html#val-to_basic

Reply:
As I a maintainer of ocaml-radare2, I can release a new version tomorrow if you send the PR. :D

Reply:
Here you go, https://github.com/fxfactorial/ocaml-radare2/pull/8, the version is bumped.

Reply:
Lets wait until it's merged in the main opam-repository: https://github.com/ocaml/opam-repository/pull/16661

Reply:
Still waiting for the opam upstream, seems like the merged version is still not upgradable now.

Reply:
It will land in the repository tonight, I suppose

Reply:
I just checked and `radare2.0.0.5` is in the opam. Please check.

Reply:
Nice! just updated the plugin to keep up with new `radare2-ocaml`